src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -126,10 +126,36 @@ public async Task<SyncTriggersResult> TrySyncTriggersAsync(bool checkHash = fals
 
         internal static bool IsSyncTriggersEnvironment(IScriptWebHostEnvironment webHostEnvironment, IEnvironment environment)
         {
-            // only want to do background sync triggers when NOT
-            // in standby mode and not running locally
-            return !environment.IsCoreToolsEnvironment() &&
-                !webHostEnvironment.InStandbyMode && environment.IsContainerReady();
+            if (environment.IsCoreToolsEnvironment())
+            {
+                // don't sync triggers when running locally or not running in app service in general
+                return false;
+            }
+
+            if (environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey) == null)
+            {
+                // We don't have the encryption key required for SetTriggers,
+                // so sync calls would fail auth anyways.
+                // This might happen in other not core tools environments for example.
+                return false;
+            }
+
+            if (webHostEnvironment.InStandbyMode)
+            {
+                // don’t sync triggers when in standby mode
+                return false;
+            }
+
+            // Windows (Dedicated/Consumption)
+            // Linux Consumption
+            if ((environment.IsAppServiceWindowsEnvironment() || environment.IsLinuxContainerEnvironment()) &&
```

Comment:
For Windows, is the `InStandbyMode` check above sufficient? Do we have to check `IsAppServiceWindowsEnvironment` here?

Reply:
I got the code from @mathewc, however (and I'm not familiar with all the placeholder logic for Windows since it's different from Linux) it seems that `WEBSITE_CONTAINER_READY` is also set on Windows and is significant. Also according to Mathew's email:

> LWAS doesn’t set the “container ready” environment variable. In both Windows (DWAS) and Linux Consumption, these environment variables are set.

I'm not sure why Windows uses "standby" AND "container ready" though.

Reply:
Thank you for the clarification. I'll follow up with @mathewc on this.

Reply:
There are two flags due to complexities in DWAS reinitialization during specialization. The second flag WEBSITE_CONTAINER_READY is set once Detours initialization is complete to indicate that runtime specialization can proceed. This was added to fix race conditions which were causing incorrect specialization (incorrect environment variables).

Reply:
Thanks for the explanation Mathew. I was curious myself too.